src/kernel-install/kernel-install.c

```diff
@@ -180,15 +185,21 @@ static int context_copy(const Context *source, Context *ret) {
         r = strdup_or_null(source->staging_area, &copy.staging_area);
         if (r < 0)
                 return r;
-        copy.plugins = strv_copy(source->plugins);
-        if (!copy.plugins)
-                return -ENOMEM;
-        copy.argv = strv_copy(source->argv);
-        if (!copy.argv)
-                return -ENOMEM;
-        copy.envp = strv_copy(source->envp);
-        if (!copy.envp)
-                return -ENOMEM;
+        if (source->plugins) {
+                copy.plugins = strv_copy(source->plugins);
+                if (!copy.plugins)
+                        return -ENOMEM;
+        }
+        if (source->argv) {
+                copy.argv = strv_copy(source->argv);
+                if (!copy.argv)
+                        return -ENOMEM;
+        }
+        if (source->envp) {
+                copy.envp = strv_copy(source->envp);
+                if (!copy.envp)
+                        return -ENOMEM;
+        }
```

Comment:
i agree with @YHNdnzj, please add a helper for this. maybe:

```c
r = strv_copy_unless_empty(source->plugins, &copy.plugins);
if (r < 0)
        return r;
```

and so on.

and then:

```c
int strv_copy_unless_empty(char **list, char ***ret) {
        assert(ret);

        if (strv_isempty(list)) {
                *ret = NULL;
                return 0; 
        }

        char **copy = strv_copy(list);
        if (!copy)
                return -ENOMEM;

        *ret = copy;
        return 1; 
}
``